client/src/main/scala/drt/client/SPAMain.scala

```diff
@@ -41,6 +63,13 @@ object SPAMain {
       }
     }
 
+    def withUrlParameters(urlParameters: Array[UrlParameter]) : TerminalPageTabLoc = {
```

Comment:
I found this very hard to read as the local defs are buried in a long line of operations

something like this clearly shows what's happening in the two cases we care about, ie None = remove, Some(value) = add

```
    def withUrlParameters(urlParameters: Array[UrlParameter]): TerminalPageTabLoc = {
      val updatedParams = urlParameters.foldLeft(queryParams) {
        case (paramsSoFar, newParam) => newParam.value match {
          case Some(newValue) => paramsSoFar.updated(newParam.name, newValue)
          case None => paramsSoFar - newParam.name
        }
      }
      copy(queryParams = updatedParams)
    }
```